src/devices/src/virtio/queue.rs

```diff
@@ -461,7 +475,16 @@ impl Queue {
             return true;
         }
 
-        if self.len(mem) != 0 {
+        let len = self.len(mem);
+        if len != 0 {
+            // The number of descriptor chain heads to process should always
+            // be smaller or equal to the queue size.
+            if len > self.actual_size() {
+                // We are choosing to interrupt execution since this could be a potential malicious driver
+                // scenario. This way we also eliminate the risk of repeatedly logging and potentially
+                // clogging the microVM through the log system.
+                panic!("The number of available virtio descriptors is greater than queue size!");
+            }
```

Comment:
```suggestion
            // We are choosing to interrupt execution since this could be a potential malicious driver
            // scenario. This way we also eliminate the risk of repeatedly logging and potentially
            // clogging the microVM through the log system.
            assert!(len > self.actual_size(), "The number of available virtio descriptors is greater than queue size!");
```

Reply:
https://github.com/firecracker-microvm/firecracker/pull/3156#discussion_r985864587